src/qtaudio/qtaudio.cc

```diff
@@ -89,18 +97,36 @@ static bool paused;
 static int last_buffered, delay_estimate;
 static timeval last_system_time;
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+static QAudioSink * output_instance = nullptr;
+#else
 static QAudioOutput * output_instance = nullptr;
+#endif
 static QIODevice * buffer_instance = nullptr;
 
 struct FormatDescriptionMap {
     int aud_format;
-
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    enum QAudioFormat::SampleFormat sample_format;
+#else
     unsigned int sample_size;
     enum QAudioFormat::SampleType sample_type;
-    enum QAudioFormat::Endian endian;    
+    enum QAudioFormat::Endian endian;
+#endif
 };
 
 static constexpr FormatDescriptionMap FormatMap[] = {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    {FMT_S16_LE, QAudioFormat::Int16},
+    {FMT_S16_BE, QAudioFormat::Int16},
```

Comment:
I think we want just the `_NE` formats here.

Reply:
You probably meant `_LE`?

Reply:
No - NE. Native-endian. It is an alias for either LE or BE depending on the target.

Reply:
So `FormatMap[]` should include for Big Endian all BE (+ FLOAT) formats and for Little Endian all LE (+ FLOAT)? Otherwise explain it with a code example please.

Reply:
Just include (for example):

```
{FMT_S16_NE, QAudioFormat::Int16},
```

Omit the explicit LE and BE formats.

Reply:
Thanks, got it now. Sometimes you think too complicated. :smiley:
For Qt 5 I have kept both signed and unsigned formats. Qt 6 supports only signed ones.